src/CloudNative.CloudEvents/Core/Validation.cs

```diff
@@ -40,6 +40,20 @@ public static void CheckArgument([DoesNotReturnIf(false)] bool condition, string
             }
         }
 
+        /// <summary>
+        /// Validates an argument-dependent condition, throwing an exception if the check fails.
+        /// </summary>
+        /// <param name="condition">The condition to validate; this method will throw an <see cref="ArgumentException"/> if this is false.</param>
+        /// <param name="paramName">The name of the parameter being validated. May be null.</param>
+        /// <param name="messageFunc">A func that returns the message to use in the exception, if one is thrown.</param>
+        public static void CheckArgument([DoesNotReturnIf(false)] bool condition, string paramName, Func<string> messageFunc)
```

Comment:
Could we keep this internal to start with? It looks like we only *need* it to be internal for now - we're not using resource strings elsewhere, and I suspect other users won't be doing so either, so I'd *prefer* to avoid exposing it.

Reply:
Sure thing 👍